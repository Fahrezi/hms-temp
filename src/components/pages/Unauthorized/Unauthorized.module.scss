.unauthorized {
  & {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    background-color: #f8d7da;
  }

  &__container {
    text-align: center;
    background: white;
    padding: 2rem;
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  }

  &__title {
    font-size: 2rem;
    color: #721c24;
    margin-bottom: 1rem;
  }

  &__message {
    font-size: 1.2rem;
    color: #721c24;
    margin-bottom: 1.5rem;
  }

  &__link {
    display: inline-block;
    padding: 0.75rem 1.5rem;
    background-color: #f5c6cb;
    color: #721c24;
    text-decoration: none;
    border-radius: 5px;
    font-weight: bold;

    &:hover {
      background-color: #f1b0b7;
    }
  }
}
