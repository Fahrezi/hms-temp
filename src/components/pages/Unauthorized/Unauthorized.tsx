import { Link } from 'react-router-dom';

import styles from './Unauthorized.module.scss';

const Unauthorized = () => {
  return (
    <div className={styles['unauthorized']}>
      <div className={styles['unauthorized__container']}>
        <h1 className={styles['unauthorized__title']}>403 - Unauthorized</h1>
        <p className={styles['unauthorized__message']}>You do not have permission to access this page.</p>
        <Link to='/' className={styles['unauthorized__link']}>
          Go to Dashboard
        </Link>
      </div>
    </div>
  );
};

export default Unauthorized;
