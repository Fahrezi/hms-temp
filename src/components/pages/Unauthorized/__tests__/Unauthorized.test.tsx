import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Unauthorized from '@/components/pages/Unauthorized';

describe('Unauthorized page Component', () => {
  it('renders the Unauthorized component', () => {
    const { container } = render(
      <MemoryRouter>
        <Unauthorized />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
