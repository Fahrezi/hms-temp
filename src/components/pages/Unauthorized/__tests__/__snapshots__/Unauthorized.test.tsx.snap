// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unauthorized page Component > renders the Unauthorized component 1`] = `
<div>
  <div
    class="_unauthorized_71b26e"
  >
    <div
      class="_unauthorized__container_71b26e"
    >
      <h1
        class="_unauthorized__title_71b26e"
      >
        403 - Unauthorized
      </h1>
      <p
        class="_unauthorized__message_71b26e"
      >
        You do not have permission to access this page.
      </p>
      <a
        class="_unauthorized__link_71b26e"
        data-discover="true"
        href="/"
      >
        Go to Dashboard
      </a>
    </div>
  </div>
</div>
`;
